lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+              Canada
+              Europe
+              Hong Kong, Philippines, S. Korea, Singapore & Taiwan
+              Japan
+              India
+            C(price_class=PriceClass_All) consists of the areas
+              United States
+              Canada
+              Europe
+              Hong Kong, Philippines, S. Korea, Singapore & Taiwan
+              Japan
+              India
+              South America
+              Australia
+      choices: ['PriceClass_100', 'PriceClass_200', 'PriceClass_All']
+      required: false
+      default: aws defaults this to 'PriceClass_All'
+
+    enabled:
+      description:
+        - A boolean value that specifies whether the distribution is enabled or
+          disabled. Applies to both distributions and streaming distributions.
+      default: 'yes'
+      choices: ['yes', 'no']
+
+    viewer_certificate:
+      description:
+        - A config element that is a complex object that specifies the
+          encryption details of the distribution. Only valid for
+          C(streaming_distribution=no).
+          Comprises the following attributes
+            C(cloudfront_default_certificate)
+            C(iam_certificate_id)
+            C(acm_certificate_arn)
+            C(ssl_support_method)
+            C(minimum_protocol_version)
+            C(certificate certificate_source)
+      required: false
+
+    restrictions:
+      description:
+        - A config element that is a complex object that describes how a
+          distribution should restrict it's content.
+          Only valid for C(streaming_distribution=no).
+          The restriction object comprises the following attributes
+            C(geo_restriction)
+              C(restriction_type)
+              C(items[])
+      required: false
+
+    s3_origin:
+      description:
+        - A config element that is used to describe the origin of a streaming
+          distribution. The s3 origin object comprises the attributes
+            C(domain_name) (defaults to C(default_s3_origin_domain_name) if not
+              specified)
+            C(origin_access_identity) (defaults to
+              C(default_streaming_s3_origin_access_identity) if not specified)
+      required: false
+
+    web_acl_id:
+      description:
+        - The id of a Web Application Firewall (WAF) Access Control List (ACL).
+          Only valid for C(streaming_distribution=no).
+      required: false
+
+    http_version:
+      description:
+        - The version of the http protocol to use for the distribution.
+          Valid for both distributions and streaming distributions.
+      choices: [ 'http1.1', 'http2' ]
+      required: false
+      default: aws defaults this to 'http2'
+
+    is_ipv6_enabled:
+      description:
+        - Determines whether IPv6 support is enabled or not.
+          Valid for both distributions and streaming distributions.
+      choices: ['yes', 'no']
+      default: 'no'
+    wait_until_processed:
```

Comment:
Use `wait`, it's standard for similar modules. Requires a configurable `wait_timeout`.

I'm not sure whether it's worse to have a default `wait` with a timeout in excess of 20 minutes so that it typically succeeds, or just no wait so that people at least knowingly set a long wait time. I'd think default to waiting off, but default to 30 minutes wait_timeout if people do set `wait`